blockchain/state_migration.go

```diff
@@ -97,7 +97,12 @@ func (bc *BlockChain) concurrentRead(db *statedb.Database, quitCh chan struct{},
 func (bc *BlockChain) migrateState(rootHash common.Hash) (returnErr error) {
 	bc.wg.Add(1)
 	defer func() {
-		bc.db.FinishStateMigration(returnErr == nil)
+		// If migration stops by quit signal, it doesn't finish migration and it it will restart again.
+		if returnErr != ErrQuitBySignal {
+			bc.mu.Lock()
```

Comment:
Perhaps, a comment is helpful?
```suggestion
			// lock to prevent from a conflict of state DB close and state DB write
			bc.mu.Lock()
```